client/src/store/index.js

```diff
@@ -32,4 +32,34 @@ const store = createStore(rootReducer, /* preloadedState, */ composeEnhancers(
       applyMiddleware(thunkMiddleware, loggerMiddleware)
     ));
 
+export const updateUnreadCount = (state, msgId) => {
+  let conversationKey = -1
+  for(let j = 0; j < state.length; j++) {
+    for(let i = 0; i < state[j].messages.length; i++){
+      if(state[j].messages[i].id === msgId){
+        conversationKey = j
+        break;
+        
+      }
+    }
+  }
```

Comment:
Is there any way you could make this more efficient, perhaps by sending the conversation ID along with the message ID?